src/logic/pluginLibraryLogic.js

```diff
@@ -61,4 +66,60 @@ export const pluginLibraryLogic = kea({
             }
         },
     }),
+
+    listeners: ({ values, actions }) => ({
+        openPlugin: async ({ pluginName }) => {
+            const pluginPath = pluginName.toLowerCase().replaceAll(' ', '-')
+            const { push } = router.actions
+            push(`/plugins/${pluginPath}`)
+
+            const { setActivePlugin } = actions
+            let plugin = values.filteredPlugins.filter((plugin) => plugin.name === pluginName)[0]
+            let markdown = `# ${plugin.name} \n ${plugin.description} \n ${pluginInstallationMd}`
+            if (plugin.url.includes('github')) {
+                const response = await window.fetch(
+                    `https://raw.githubusercontent.com/${plugin.url.split('hub.com/')[1]}/main/README.md`
+                )
+                if (response.status === 200) {
+                    markdown = await response.text()
+                }
+            }
+            if (!markdown.includes('Installation')) {
+                markdown += pluginInstallationMd
+            }
+            plugin['markdown'] = markdown.split(/!\[.*\]\(.*\)/).join('')
+            plugin['imageSrc'] = getPluginImageSrc(plugin)
+
+            setActivePlugin(plugin)
+        },
+        closeModal: () => {
+            if (window.location.pathname !== '/plugins') {
+                const { push } = router.actions
+                push('/plugins')
+            }
+        },
+        [router.actions.locationChanged]: ({ pathname }) => {
+            if (pathname === '/plugins') {
+                actions.closeModal()
+            }
+        },
```

Comment:
Don't worry about the URL in listeners (not here nor in openPlugin/closeModal). Use the special `urlToAction` and `actionToUrl` thingies found in [kea-router](https://kea.js.org/docs/plugins/router/)